Code/Editor/AzAssetBrowser/AzAssetBrowserWindow.cpp

```diff
@@ -786,32 +790,48 @@ static void ExpandTreeToIndex(QTreeView* treeView, const QModelIndex& index)
 
 void AzAssetBrowserWindow::SelectAsset(const QString& assetPath)
 {
-    QModelIndex index = m_assetBrowserModel->findIndex(assetPath);
-    if (index.isValid())
+    if (ed_useWIPAssetBrowserDesign)
     {
-        m_ui->m_searchWidget->ClearTextFilter();
-        m_ui->m_searchWidget->ClearTypeFilter();
-
-        // Queue the expand and select stuff, so that it doesn't get processed the same
-        // update as the search widget clearing - something with the search widget clearing
-        // interferes with the update from the select and expand, and if you don't
-        // queue it, the tree doesn't expand reliably.
-
         QTimer::singleShot(
-            0, this,
-            [this, filteredIndex = index]
+            0,
+            this,
+            [this, assetPath]
             {
-                // the treeview has a filter model so we have to backwards go from that
-                QModelIndex index = m_filterModel->mapFromSource(filteredIndex);
+                m_ui->m_searchWidget->ClearTextFilter();
+                m_ui->m_searchWidget->ClearTypeFilter();
 
-                QTreeView* treeView = m_ui->m_assetBrowserTreeViewWidget;
-                ExpandTreeToIndex(treeView, index);
+                m_ui->m_assetBrowserTreeViewWidget->SelectFileAtPathAfterUpdate(assetPath.toUtf8().data());
+            });
+    }
+    else
+    {
+        QModelIndex index = m_assetBrowserModel->findIndex(assetPath);
+        if (index.isValid())
+        {
+            m_ui->m_searchWidget->ClearTextFilter();
+            m_ui->m_searchWidget->ClearTypeFilter();
+
+            // Queue the expand and select stuff, so that it doesn't get processed the same
+            // update as the search widget clearing - something with the search widget clearing
+            // interferes with the update from the select and expand, and if you don't
+            // queue it, the tree doesn't expand reliably.
+
+            QTimer::singleShot(
+                0,
+                this,
+                [this, filteredIndex = index]
+                {
+                    // the treeview has a filter model so we have to backwards go from that
+                    QModelIndex modelIndex = m_filterModel->mapFromSource(filteredIndex);
 
-                treeView->scrollTo(index);
-                treeView->setCurrentIndex(index);
+                    QTreeView* treeView = m_ui->m_assetBrowserTreeViewWidget;
+                    ExpandTreeToIndex(treeView, modelIndex);
 
-                treeView->selectionModel()->select(index, QItemSelectionModel::ClearAndSelect);
-            });
+                    treeView->scrollTo(modelIndex);
+                    treeView->setCurrentIndex(modelIndex);
+                    treeView->selectionModel()->select(modelIndex, QItemSelectionModel::ClearAndSelect);
```

Comment:
I know this is in the old AB, but it seems like this would break multi-select in the tree view

Reply:
It's in the old AB and the existing code.